### ISSUE TITLE
Allow jobs to define their triggers only if both resources have a new version

### ISSUE THREAD

**Issue by cjcjameson** [concourse/concourse#604]
As a team, we have heavy-weight tests at the end of our pipeline. We want to trigger these tests when "Time resource triggers for a (24hour) nightly build" AND "there is a new SHA / artifact to test against". We don't want to trigger every night if the code/artifact haven't changed, and we don't want to trigger on every commit.

You could also phrase it like: "trigger the job every day at 11pm but skip/abort if the latest git SHA is more than 24 hours old". One rudimentary implementation would be to trigger based on time, but then have logic in a quick task at the beginning of the job that verifies the recency criterion on the git commit.

But this latter phrasing also directs our attention to the [get-version](http://concourse.ci/get-step.html#get-version) feature. A time-specific implementation might be to add an option in the `version:` field to let you specify a time frequency / window. This depends on resources (such as git) reporting or being able to determine what is recent (e.g. not commit-time but rather merge-time).

But perhaps there's a more general implementation that doesn't overload the `version:` field... one that works across pipeline get steps. Perhaps as a feature of an aggregate? For example: `aggregate:` can take a flag/property `trigger_all_only: true` to change the triggering logic among the constituent `get`s. Although I'm guessing that y'all like the cleanliness of `aggregate:` only being used to semantically describe concurrency control.

This also could apply to larger sets of 3+ resources, but using the example of two resources here for simplicity

**Comment by ggeorgiev** [comment]
Actually, this could be achieved without a change with an extra throttle job. For example, if you have:

    - name: task
      plan:
      - aggregate:
        - get: resource1
          passed: [job1]
          trigger: true

        - get: resource2
          passed: [job2]
          trigger: true

        - task: do_something
          ...

All you need is to replace it with that:

    - name: throttle
      plan:
      - aggregate:
        - get: throttle_time
          trigger: true

        - get: resource1
          passed: [job1]

        - get: resource2
          passed: [job2]


    - name: task
      plan:
      - aggregate:
        - get: resource1
          passed: [throttle]
          trigger: true

        - get: resource2
          passed: [throttle]
          trigger: true

        - task: do_something
          ...

**Comment by cjcjameson** [comment]
Agreed, that looks like the best workaround -- thank you for laying out the minimal pattern required! It's a cool example of what a Job with neither tasks nor outputs could be.

**Comment by taylorsilva** [comment]
# Issue Bankruptcy

This issue is being closed as part of 2025's **Issue Bankruptcy** declaration. [See here for details](https://blog.concourse-ci.org/posts/2025-01-06-concourse-2024-review/)

If you think this issue still has merit then feel free to open a new issue and reference this one.
